app/controllers/returns/ManufacturedPlasticPackagingController.scala

```diff
@@ -35,51 +36,49 @@ import scala.concurrent.{ExecutionContext, Future}
 class ManufacturedPlasticPackagingController @Inject() (
   override val messagesApi: MessagesApi,
   cacheConnector: CacheConnector,
-  navigator: Navigator,
   identify: IdentifierAction,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
-  form: ManufacturedPlasticPackagingFormProvider,
+  formProvider: ManufacturedPlasticPackagingFormProvider,
   val controllerComponents: MessagesControllerComponents,
-  view: ManufacturedPlasticPackagingView
+  view: ManufacturedPlasticPackagingView,
+  returnsNavigator: ReturnsJourneyNavigator
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(mode: Mode): Action[AnyContent] =
-    (identify andThen getData andThen requireData).async {
+    (identify andThen getData andThen requireData) {
       implicit request =>
-        val preparedForm = request.userAnswers.get(ManufacturedPlasticPackagingPage) match {
-          case None        => form()
-          case Some(value) => form().fill(value)
-        }
+        val preparedForm = request.userAnswers.fill(ManufacturedPlasticPackagingPage, formProvider())
 
         request.userAnswers.get[TaxReturnObligation](ObligationCacheable) match {
-          case Some(obligation) => Future.successful(Ok(view(preparedForm, mode, obligation)))
-          case None             => Future.successful(Redirect(controllers.routes.IndexController.onPageLoad))
+          case Some(obligation) => Ok(view(preparedForm, mode, obligation))
+          case None => Redirect(controllers.routes.IndexController.onPageLoad)
         }
     }
 
   def onSubmit(mode: Mode): Action[AnyContent] =
     (identify andThen getData andThen requireData).async {
       implicit request =>
         val pptId: String = request.pptReference
+        val userAnswers = request.userAnswers
```

Comment:
non a blocker, but this could be inline as they only needed if successful. These would not be needed to retrieved if we have a validation error